packages/web-frontend/src/utils/periodGranularities.js

```diff
@@ -164,6 +164,13 @@ export const momentToDateString = (date, granularity, format) =>
  * @param {*} periodGranularity
  * @param {*} defaultTimePeriod
  */
+
+const validateDateString = date => {
+  if (!moment(date).isValid()) {
```

Comment:
Good test about an invalid date! That's even more important than what I was thinking, which is just around using a consistent format to represent the date. `moment()` will accept a wide range of formats, but with some it will be doing a "best guess", and anyway I think it'd be good for us to use a single consistent format. I think you could get this format checking by adding a single format parsing argument, but not totally sure
```suggestion
  if (!moment(date, 'YYYY-MM-DD').isValid()) {
```